### PR TITLE
eternal: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **eternal/1.0.1**

Eternal is constexpr container library.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]
LGTM 👌

File: recipes/eternal/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, get, copy
```
Comment (prince-chrismc) [inline recipes/eternal/all/conanfile.py:3]: Linter

```suggestion
from conan.tools.files import get, copy
```

File: recipes/eternal/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "msvc": "1910",
```
Comment (prince-chrismc) [inline recipes/eternal/all/conanfile.py:31]: I thought this was 3 digits 🤔 https://github.com/conan-io/conan-center-index/blob/a932b2aa205b9aebfc728fee3914675134e02eb0/recipes/argparse/all/conanfile.py#L34

Comment (toge) [reply recipes/eternal/all/conanfile.py:31]: @prince-chrismc 
I misunderstanding msvc version.
Thank you.

Comment (toge) [reply recipes/eternal/all/conanfile.py]: fixed

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**